ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -275,6 +275,75 @@ objects:
         /bin/openshift-install --dir=/tmp/artifacts/installer --log-level=debug create cluster &
         wait "$!"
 
+    # Stream "journal -u bootkube" and "kubectl get all -a --all-namespaces" into artifact
+    - name: stream-bootstrapping
+      image: ${IMAGE_INSTALLER}
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-profile
+        mountPath: /etc/openshift-installer
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        mkdir -p .ssh
+        chmod 700 .ssh
+        cp /etc/openshift-installer/ssh-privatekey .ssh/id_rsa
+        cp /etc/openshift-installer/ssh-publickey .ssh/id_rsa.pub
+
+        function stream-bootkube () {
+          ip=${1}
+          while true; do
+            ssh -o "StrictHostKeyChecking=no" core@${ip} sudo journal -u bootkube 2>&1 >> /tmp/artifacts/bootstrap/bootkube.log
+            date >> /tmp/artifacts/bootstrap/bootkube.log
+            sleep 5
+          done
+        }
+
+        function stream-kubectl-get-all () {
+          ip=${1}
+          while true; do
+            ssh -o "StrictHostKeyChecking=no" core@${ip} sudo kubectl get all --kubeconfig=/var/opt/tectonic/auth/kubeconfig -a --all-namespaces -w 2>&1 >> /tmp/artifacts/bootstrap/kubectl-get-all.log
+            date >> /tmp/artifacts/bootstrap/kubectl-get-all.log
+            sleep 5
+          done
+        }
+
+        function start-streams () {
+          # wait for terraform to start up
+          while [ ! -f /tmp/artifacts/installer/terraform.tfstate ]; do sleep 5; done
+
+          # wait for the bootstrap node to show up with an IP
+          ip=""
+          while [ -z "${ip}" ]; do
+            ip=$(terraform state show -state=terraform.tfstate module.bootstrap.aws_instance.bootstrap | sed -n 's/^public_ip *= *//p')
+          done
+
+          # try to login
+          while ! ssh -o "StrictHostKeyChecking=no" core@${ip} /bin/true; do
+            sleep 5
+          done
+
+          # start streaming
+          mkdir -p /tmp/artifacts/bootstrap
+          stream-bootkube ${ip} &
+          stream-kubectl-get-all ${ip} &
+        }
+
+        start-streams
+        for i in `seq 1 120`; do
+          if [[ -f /tmp/shared/exit ]]; then
+            exit 0
+          fi
+          sleep 60 & wait
```

Comment:
I don't understand the backgrounded sleep here.

```console
$ (sleep 10 && echo 'done sleeping') & (wait && echo 'done waiting')
done waiting  # this shows up very quickly
done sleeping  # this is delayed by 10 seconds
```

Why have a non-blocking `sleep`?  I'd expect `sleep` here with a trailing `kill`:

```
for i in $(seq 1 120); do
  if [[ -f /tmp/shared/exit ]]; then
    break
  fi
  sleep 60
done
kill-streams
```

where `kill-streams` had internal `wait` calls (possibly using explicit PIDs).

Reply:
the loop with the sleep is copied from the teardown script below.

Reply:
the kill-streams is not necessary due to the TERM trap